### PR TITLE
hw_9_corrections

### PR DESCRIPTION
**Pull request by Oleksiy-Yurchenko**
Uploaded hw_9 with corrections.

### CODE REVIEW

File: hw_9_corr.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+class Warehouse:
+    def __init__(self):
+        self.stock = []
+
+    def find_material(self, name):
+        if self.stock:
+            for material in self.stock:
+                if material.name == name:
+                    return material
+        return None
```
Comment (mmykhailenko) [inline hw_9_corr.py]: no sense in this line, could be removed

File: hw_9_corr.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+    def add_material(self, name, details, quantity, availability, price):
+        material = Material(name, details, quantity, availability, price)
+        self.stock.append(material)
+
+    def remove_material(self, material):
+        material = self.find_material(material)
+        if material:
+            self.stock.remove(material)
+
+    def current_quantity(self, material):
```
Comment (mmykhailenko) [inline hw_9_corr.py]: name of this method should be `print_current_quantity` or `get_current_quantity`

File: hw_9_corr.py [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        material = self.find_material(material)
+        if material:
+            print('Current stock of {0} is {1}'.format(material.name, material.quantity))
+        else:
+            print('Material does not exist.')
+
+    def current_quantity_all(self):
```
Comment (mmykhailenko) [inline hw_9_corr.py]: name of this method should be `print_current_quantity_all` or `get_current_quantity_all`

File: hw_9_corr.py [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        if self.stock:
+            for material in self.stock:
+                self.current_quantity(material.name)
+        else:
+            print('Warehouse is empty.')
+
+
+class Category(Warehouse):
```
Comment (mmykhailenko) [inline hw_9_corr.py]: why we need here Warehouse as a parent?

File: hw_9_corr.py [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    def print_order(self):
+        id = 0
+        print('Order ID: ', self.order_id)
+        print('Date purchased: ', self.date_purchased)
+        for material in self.items:
+            id += 1
+            print(id, 'Material: ', material.name, 'Quantity: ', self.items[material])
+
+
+order_id = 0
```
Comment (mmykhailenko) [inline hw_9_corr.py]: why order_id as a GLOBAL? this var should be only for Order class

File: hw_9_corr.py [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+                print(id, 'Material: ', material.name, 'Quantity: ', self.content[material], 'Price: ', material.price,
+                      'Total: ', self.content[material] *  material.price)
+            print('Grand Total: ', self.total)
+
+
+class Order(Basket):
+    def __init__(self):
+        self.items = {}
+        global order_id
+        order_id += 1
```
Comment (mmykhailenko) [inline hw_9_corr.py]: move to method like `_generate_order_id`

File: hw_9_corr.py [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+        print('Date purchased: ', self.date_purchased)
+        for material in self.items:
+            id += 1
+            print(id, 'Material: ', material.name, 'Quantity: ', self.items[material])
+
+
+order_id = 0
+
+
+class OrderManager(Basket):
```
Comment (mmykhailenko) [inline hw_9_corr.py]: why Basket as parent class here?

Comment (Oleksiy-Yurchenko) [reply hw_9_corr.py]: Parent class Basket has been remove.

File: hw_9_corr.py (hunk 6 above)
Comment (Oleksiy-Yurchenko) [reply hw_9_corr.py]: Method _generate_order_id has been created.

File: hw_9_corr.py (hunk 5 above)
Comment (Oleksiy-Yurchenko) [reply hw_9_corr.py]: var order_id has been moved to the Order class and set as a class atribute,

File: hw_9_corr.py (hunk 4 above)
Comment (Oleksiy-Yurchenko) [reply hw_9_corr.py]: Parent class Warehouse has been removed from Category Class.

File: hw_9_corr.py (hunk 3 above)
Comment (Oleksiy-Yurchenko) [reply hw_9_corr.py]: Amended accordingly.

File: hw_9_corr.py (hunk 2 above)
Comment (Oleksiy-Yurchenko) [reply hw_9_corr.py]: Method current_quantity has been renamed to print_current_quantity.

File: hw_9_corr.py (hunk 1 above)
Comment (Oleksiy-Yurchenko) [reply hw_9_corr.py]: The line 'return None' has been removed.

**Review by mmykhailenko — APPROVED** [review]
LGTM

**Pull request merged**